### DISCUSSION TITLE
The Kernel crashed while executing code

### DISCUSSION THREAD

**Discussion by tun36782** [microsoft/vscode-jupyter#15578] [Questions and Answers] [answered]
my kernel log says this and I can't seems to understand what's wrong:
I'm using the m2 MacBook Air:
12:19:52.740 [info] Restart requested ~/Documents/University/Robotic AI/hand-gesture-recognition-mediapipe-main/keypoint_classification_EN.ipynb
12:19:52.741 [info] End cell 25 execution after 0s, completed @ undefined, started @ undefined
12:19:52.741 [info] End cell 26 execution after 0s, completed @ undefined, started @ undefined
12:19:52.741 [info] End cell 27 execution after 0s, completed @ undefined, started @ undefined
12:19:52.741 [info] End cell 28 execution after 0s, completed @ undefined, started @ undefined
12:19:52.741 [info] End cell 29 execution after 0s, completed @ undefined, started @ undefined
12:19:52.742 [warn] Cancel all remaining cells due to cancellation or failure in execution
12:19:52.779 [warn] Failed to get activated env vars for /usr/bin/python3 in 34ms
12:19:52.781 [info] Process Execution: /usr/bin/python3 -c "import ipykernel; print(ipykernel.__version__); print("5dc3a68c-e34e-4080-9c3e-2a532b2ccb4d"); print(ipykernel.__file__)"
12:19:52.782 [info] Process Execution: /usr/bin/python3 -m ipykernel_launcher --f=~/Library/Jupyter/runtime/kernel-v2-26408cP3WcPX0Ne1s.json
    > cwd: ~/Documents/University/Robotic AI/hand-gesture-recognition-mediapipe-main
12:19:53.224 [info] Restarted 3f89cccc-5649-4df1-8d6f-8dab21b447e6
12:19:53.835 [info] Handle Execution of Cells 0,2,4,6,7,8,10,11,12,13,15,16,17,18,20,22,23,25,26,27,28,29 for ~/Documents/University/Robotic AI/hand-gesture-recognition-mediapipe-main/keypoint_classification_EN.ipynb
12:19:53.841 [info] Kernel acknowledged execution of cell 0 @ 1713676793841
12:19:57.984 [info] End cell 0 execution after 4.143s, completed @ 1713676797984, started @ 1713676793841
12:19:57.987 [info] Kernel acknowledged execution of cell 2 @ 1713676797986
12:19:57.988 [info] End cell 2 execution after 0.002s, completed @ 1713676797988, started @ 1713676797986
12:19:57.990 [info] Kernel acknowledged execution of cell 4 @ 1713676797990
12:19:57.991 [info] End cell 4 execution after 0.001s, completed @ 1713676797991, started @ 1713676797990
12:19:57.994 [info] Kernel acknowledged execution of cell 6 @ 1713676797994
12:19:58.028 [info] End cell 6 execution after 0.034s, completed @ 1713676798028, started @ 1713676797994
12:19:58.031 [info] Kernel acknowledged execution of cell 7 @ 1713676798031
12:19:58.042 [info] End cell 7 execution after 0.011s, completed @ 1713676798042, started @ 1713676798031
12:19:58.046 [info] Kernel acknowledged execution of cell 8 @ 1713676798046
12:19:58.048 [info] End cell 8 execution after 0.002s, completed @ 1713676798048, started @ 1713676798046
12:19:58.086 [info] Kernel acknowledged execution of cell 10 @ 1713676798086
12:19:58.106 [info] End cell 10 execution after 0.02s, completed @ 1713676798106, started @ 1713676798086
12:19:58.110 [info] Kernel acknowledged execution of cell 11 @ 1713676798110
12:19:58.116 [info] End cell 11 execution after 0.006s, completed @ 1713676798116, started @ 1713676798110
12:19:58.118 [info] Kernel acknowledged execution of cell 12 @ 1713676798118
12:19:58.120 [info] End cell 12 execution after 0.002s, completed @ 1713676798120, started @ 1713676798118
12:19:58.122 [info] Kernel acknowledged execution of cell 13 @ 1713676798121
12:19:58.128 [info] End cell 13 execution after 0.007s, completed @ 1713676798128, started @ 1713676798121
12:19:58.131 [info] Kernel acknowledged execution of cell 15 @ 1713676798130
12:20:07.469 [info] End cell 15 execution after 9.339s, completed @ 1713676807469, started @ 1713676798130
12:20:07.473 [info] Kernel acknowledged execution of cell 16 @ 1713676807473
12:20:07.510 [info] End cell 16 execution after 0.037s, completed @ 1713676807510, started @ 1713676807473
12:20:07.512 [info] Kernel acknowledged execution of cell 17 @ 1713676807512
12:20:07.545 [info] End cell 17 execution after 0.033s, completed @ 1713676807545, started @ 1713676807512
12:20:07.548 [info] Kernel acknowledged execution of cell 18 @ 1713676807548
12:20:07.595 [info] End cell 18 execution after 0.047s, completed @ 1713676807595, started @ 1713676807548
12:20:07.598 [info] Kernel acknowledged execution of cell 20 @ 1713676807598
12:20:08.071 [info] End cell 20 execution after 0.473s, completed @ 1713676808071, started @ 1713676807598
12:20:08.075 [info] Kernel acknowledged execution of cell 22 @ 1713676808074
12:20:08.099 [info] End cell 22 execution after 0.025s, completed @ 1713676808099, started @ 1713676808074
12:20:08.116 [info] Kernel acknowledged execution of cell 23 @ 1713676808116
12:20:08.929 [error] Disposing session as kernel process died ExitCode: undefined, Reason: 
12:20:08.929 [info] Dispose Kernel process 27117.
12:20:08.944 [info] End cell 23 execution after -1713676808.116s, completed @ undefined, started @ 1713676808116

I tried to train the model from this video clip[ https://www.youtube.com/watch?v=a99p_fAr6e4&list=WL&index=41&t=675s ] on 53:30 minute mark and I am running the file called "Keypoint_Classification_EN.ipynb"

send help plz :(

**Comment by DonJayamanne** [comment]
> minute mark and I am running the file called "Keypoint_Classification_EN.ipynb"

Majority of the time this is caused by incorrectly installed Python packages,

Based on the logs, the kernel crashes when running the code in cell 23, some code in there is whats causing the kernel to crash (will give you a hint to the package etc),
In the past we've found incorrectly installed tensorflow packages cause such issues.

Please can you run the same notebook in Jupyter Lab/Notebook via the Terminal outside VS Code and share the output from the terminal. I suspect it will crash in there as well, please share the output from the Terminal, hopefully that will cotnain some additional information.

**Reply by tun36782** [reply]
this is the cell 23 and this is what it says in vs code( it is the same as the ones I run on jupiter notebook via anaconda)
<img width="1470" alt="Screenshot 2024-04-23 at 09 31 03" src="https://github.com/microsoft/vscode-jupyter/assets/162990465/1ea3c671-b44e-4eff-bdea-2bcf8f8e57fb">

**Reply by tun36782** [reply]
Here is my pip list 
##################################
Package                           Version
--------------------     -------------------
absl-py                           2.1.0
aiobotocore                       2.7.0
aiohttp                           3.9.3
aioitertools                      0.7.1
aiosignal                         1.2.0
alabaster                         0.7.12
altair                            5.0.1
anaconda-anon-usage               0.4.3
anaconda-catalogs                 0.2.0
anaconda-client                   1.12.3
anaconda-cloud-auth               0.1.4
anaconda-navigator                2.5.2
anaconda-project                  0.11.1
anyio                             4.2.0
appdirs                           1.4.4
applaunchservices                 0.3.0
appnope                           0.1.2
appscript                         1.1.2
archspec                          0.2.3
argon2-cffi                       21.3.0
argon2-cffi-bindings              21.2.0
arrow                             1.2.3
astroid                           2.14.2
astropy                           5.3.4
asttokens                         2.0.5
astunparse                        1.6.3
async-lru                         2.0.4
atomicwrites                      1.4.0
attrs                             23.1.0
Automat                           20.2.0
autopep8                          1.6.0
azure-cognitiveservices-speech    1.36.0
Babel                             2.11.0
backports.functools-lru-cache     1.6.4
backports.tempfile                1.0
backports.weakref                 1.0.post1
bcrypt                            3.2.0
beautifulsoup4                    4.12.2
binaryornot                       0.4.4
black                             23.11.0
bleach                            4.1.0
blinker                           1.6.2
bokeh                             3.3.4
boltons                           23.0.0
botocore                          1.31.64
Bottleneck                        1.3.7
Brotli                            1.0.9
cachetools                        4.2.2
certifi                           2024.2.2
cffi                              1.16.0
chardet                           4.0.0
charset-normalizer                2.0.4
click                             8.1.7
cloudpickle                       2.2.1
clyent                            1.2.2
colorama                          0.4.6
colorcet                          3.0.1
comm                              0.2.1
conda                             24.3.0
conda-build                       24.1.2
conda-content-trust               0.2.0
conda_index                       0.4.0
conda-libmamba-solver             24.1.0
conda-pack                        0.6.0
conda-package-handling            2.2.0
conda_package_streaming           0.9.0
conda-repo-cli                    1.0.75
conda-token                       0.4.0
conda-verify                      3.4.2
constantly                        23.10.4
contourpy                         1.2.0
cookiecutter                      2.5.0
cryptography                      42.0.2
cssselect                         1.2.0
cvzone                            1.6.1
cycler                            0.11.0
cytoolz                           0.12.2
dask                              2023.11.0
datashader                        0.16.0
debugpy                           1.6.7
decorator                         5.1.1
defusedxml                        0.7.1
diff-match-patch                  20200713
dill                              0.3.7
distributed                       2023.11.0
distro                            1.8.0
docstring-to-markdown             0.11
docutils                          0.18.1
entrypoints                       0.4
et-xmlfile                        1.1.0
executing                         0.8.3
fastjsonschema                    2.16.2
filelock                          3.13.1
flake8                            6.0.0
Flask                             2.2.5
flatbuffers                       24.3.25
fonttools                         4.25.0
frozenlist                        1.4.0
fsspec                            2023.10.0
future                            0.18.3
gast                              0.5.4
gensim                            4.3.0
gitdb                             4.0.7
GitPython                         3.1.37
gmpy2                             2.1.2
google-pasta                      0.2.0
greenlet                          3.0.1
grpcio                            1.62.1
h5py                              3.11.0
HeapDict                          1.0.1
holoviews                         1.18.3
hvplot                            0.9.2
hyperlink                         21.0.0
idna                              3.4
imagecodecs                       2023.1.23
imageio                           2.33.1
imagesize                         1.4.1
imbalanced-learn                  0.11.0
importlib-metadata                7.0.1
incremental                       22.10.0
inflection                        0.5.1
iniconfig                         1.1.1
intake                            0.6.8
intervaltree                      3.1.0
ipykernel                         6.28.0
ipython                           8.20.0
ipython-genutils                  0.2.0
ipywidgets                        7.6.5
isort                             5.9.3
itemadapter                       0.3.0
itemloaders                       1.1.0
itsdangerous                      2.0.1
jaraco.classes                    3.2.1
jax                               0.4.26
jaxlib                            0.4.26
jedi                              0.18.1
jellyfish                         1.0.1
Jinja2                            3.1.3
jmespath                          1.0.1
joblib                            1.2.0
json5                             0.9.6
jsonpatch                         1.32
jsonpointer                       2.1
jsonschema                        4.19.2
jsonschema-specifications         2023.7.1
jupyter                           1.0.0
jupyter_client                    8.6.0
jupyter-console                   6.6.3
jupyter_core                      5.5.0
jupyter-events                    0.8.0
jupyter-lsp                       2.2.0
jupyter_server                    2.10.0
jupyter_server_terminals          0.4.4
jupyterlab                        4.0.11
jupyterlab-pygments               0.1.2
jupyterlab_server                 2.25.1
jupyterlab-widgets                3.0.9
keras                             3.2.0
keras-nightly                     3.3.0.dev2024042103
keyring                           23.13.1
kiwisolver                        1.4.4
lazy_loader                       0.3
lazy-object-proxy                 1.6.0
lckr_jupyterlab_variableinspector 3.1.0
libarchive-c                      2.9
libclang                          18.1.1
libmambapy                        1.5.6
linkify-it-py                     2.0.0
llvmlite                          0.42.0
lmdb                              1.4.1
locket                            1.0.0
lxml                              4.9.3
lz4                               4.3.2
Markdown                          3.4.1
markdown-it-py                    2.2.0
MarkupSafe                        2.1.3
matplotlib                        3.8.0
matplotlib-inline                 0.1.6
mccabe                            0.7.0
mdit-py-plugins                   0.3.0
mdurl                             0.1.0
media                             0.1.6
mediapipe                         0.10.9
mediapipe-silicon                 0.9.2.1
menuinst                          2.0.2
mistune                           2.0.4
ml-dtypes                         0.3.2
more-itertools                    10.1.0
mpmath                            1.3.0
msgpack                           1.0.3
multidict                         6.0.4
multipledispatch                  0.6.0
munkres                           1.1.4
mypy                              1.8.0
mypy-extensions                   1.0.0
namex                             0.0.8
navigator-updater                 0.4.0
nbclient                          0.8.0
nbconvert                         7.10.0
nbformat                          5.9.2
nest-asyncio                      1.6.0
networkx                          3.1
nltk                              3.8.1
notebook                          7.0.8
notebook_shim                     0.2.3
numba                             0.59.0
numexpr                           2.8.7
numpy                             1.26.4
numpydoc                          1.5.0
opencv-contrib-python             4.9.0.80
opencv-python                     4.9.0.80
openpyxl                          3.0.10
opt-einsum                        3.3.0
optree                            0.11.0
overrides                         7.4.0
packaging                         23.2
pandas                            2.1.4
pandocfilters                     1.5.0
panel                             1.3.8
param                             2.0.2
parsel                            1.8.1
parso                             0.8.3
partd                             1.4.1
pathlib                           1.0.1
pathspec                          0.10.3
patsy                             0.5.3
pexpect                           4.8.0
pickleshare                       0.7.5
pillow                            10.2.0
pip                               24.0
pipe                              2.2
pkce                              1.0.3
pkginfo                           1.9.6
platformdirs                      3.10.0
plotly                            5.9.0
pluggy                            1.0.0
ply                               3.11
prometheus-client                 0.14.1
prompt-toolkit                    3.0.43
Protego                           0.1.16
protobuf                          3.20.3
psutil                            5.9.0
ptyprocess                        0.7.0
pure-eval                         0.2.2
py-cpuinfo                        9.0.0
pyarrow                           14.0.2
pyasn1                            0.4.8
pyasn1-modules                    0.2.8
pycodestyle                       2.10.0
pycosat                           0.6.6
pycparser                         2.21
pyct                              0.5.0
pycurl                            7.45.2
pydantic                          1.10.12
pydeck                            0.8.0
PyDispatcher                      2.0.5
pydocstyle                        6.3.0
pyerfa                            2.0.0
pyflakes                          3.0.1
Pygments                          2.15.1
PyJWT                             2.4.0
pylance                           0.10.11
pylint                            2.16.2
pylint-venv                       2.3.0
pyls-spyder                       0.4.0
pyobjc-core                       9.0
pyobjc-framework-Cocoa            9.0
pyobjc-framework-CoreServices     9.0
pyobjc-framework-FSEvents         9.0
pyodbc                            5.0.1
pyOpenSSL                         24.0.0
pyparsing                         3.0.9
PyQt5                             5.15.10
PyQt5-sip                         12.13.0
PyQtWebEngine                     5.15.6
PySocks                           1.7.1
pytest                            7.4.0
python-dateutil                   2.8.2
python-dotenv                     0.21.0
python-json-logger                2.0.7
python-lsp-black                  1.2.1
python-lsp-jsonrpc                1.0.0
python-lsp-server                 1.7.2
python-slugify                    5.0.2
python-snappy                     0.6.1
pytoolconfig                      1.2.6
pytz                              2023.3.post1
pyviz_comms                       3.0.0
pywavelets                        1.5.0
PyYAML                            6.0.1
pyzmq                             25.1.2
QDarkStyle                        3.0.2
qstylizer                         0.2.2
QtAwesome                         1.2.2
qtconsole                         5.4.2
QtPy                              2.4.1
queuelib                          1.6.2
referencing                       0.30.2
regex                             2023.10.3
requests                          2.31.0
requests-file                     1.5.1
requests-toolbelt                 1.0.0
rfc3339-validator                 0.1.4
rfc3986-validator                 0.1.1
rich                              13.3.5
rope                              1.7.0
rpds-py                           0.10.6
Rtree                             1.0.1
ruamel.yaml                       0.17.21
ruamel-yaml-conda                 0.17.21
s3fs                              2023.10.0
scikit-image                      0.22.0
scikit-learn                      1.2.2
scipy                             1.11.4
Scrapy                            2.8.0
seaborn                           0.12.2
semver                            2.13.0
Send2Trash                        1.8.2
service-identity                  18.1.0
setuptools                        69.2.0
sip                               6.7.12
six                               1.16.0
smart-open                        5.2.1
smmap                             4.0.0
sniffio                           1.3.0
snowballstemmer                   2.2.0
sortedcontainers                  2.4.0
sounddevice                       0.4.6
soupsieve                         2.5
Sphinx                            5.0.2
sphinxcontrib-applehelp           1.0.2
sphinxcontrib-devhelp             1.0.2
sphinxcontrib-htmlhelp            2.0.0
sphinxcontrib-jsmath              1.0.1
sphinxcontrib-qthelp              1.0.3
sphinxcontrib-serializinghtml     1.1.5
spyder                            5.4.3
spyder-kernels                    2.4.4
SQLAlchemy                        2.0.25
stack-data                        0.2.0
statsmodels                       0.14.0
streamlit                         1.30.0
sympy                             1.12
tables                            3.9.2
tabulate                          0.9.0
tb-nightly                        2.17.0a20240420
tblib                             1.7.0
tenacity                          8.2.2
tensorboard                       2.16.2
tensorboard-data-server           0.7.2
tensorflow                        2.16.1
tensorflow-io-gcs-filesystem      0.36.0
tensorflow-macos                  2.16.1
tensorflow-metal                  1.1.0
termcolor                         2.4.0
terminado                         0.17.1
text-unidecode                    1.3
textdistance                      4.2.1
tf-nightly                        2.17.0.dev20240420
tflite                            2.3.0
threadpoolctl                     2.2.0
three-merge                       0.1.1
tifffile                          2023.4.12
tinycss2                          1.2.1
tldextract                        3.2.0
toml                              0.10.2
tomlkit                           0.11.1
toolz                             0.12.0
torch                             2.2.2
tornado                           6.3.3
tqdm                              4.65.0
traitlets                         5.7.1
truststore                        0.8.0
Twisted                           23.10.0
typing_extensions                 4.9.0
tzdata                            2023.3
tzlocal                           2.1
uc-micro-py                       1.0.1
ujson                             5.4.0
Unidecode                         1.2.0
urllib3                           2.0.7
validators                        0.18.2
w3lib                             2.1.2
watchdog                          2.1.6
wcwidth                           0.2.5
webencodings                      0.5.1
websocket-client                  0.58.0
Werkzeug                          2.2.3
whatthepatch                      1.0.2
wheel                             0.41.2
widgetsnbextension                3.5.2
wrapt                             1.14.1
wurlitzer                         3.0.2
xarray                            2023.6.0
xlwings                           0.29.1
xyzservices                       2022.9.0
yapf                              0.31.0
yarl                              1.9.3
zict                              3.0.0
zipp                              3.17.0
zope.interface                    5.4.0
zstandard                         0.19.0

**Reply by DonJayamanne** [reply, marked as answer]
I'm 100% certain this is caused by an incorrect installation of tensorflow & or related packages.
My suggestion is to reach out on stack overflow for this,

> Please can you run the same notebook in Jupyter Lab/Notebook via the Terminal outside VS Code and share the output from the terminal. I suspect it will crash in there as well, please share the output from the Terminal, hopefully that will cotnain some additional information.

As mentioned earlier, if you run this exact same code agains the same Python environment it will crash outside VS Code as well, meaning its not an issue with VS Code
